kotlin/workflow-core/src/main/java/com/squareup/workflow/StatelessWorkflow.kt

```diff
@@ -16,15 +16,78 @@
 package com.squareup.workflow
 
 /**
- * A [Workflow] that has no state.
+ * A composable object that can [handle events][WorkflowContext.onEvent],
+ * [delegate to children][WorkflowContext.compose], and [subscribe][onReceive] to arbitrary streams
+ * from the outside world.
+ *
+ * The basic purpose of a `Workflow` is to take some [input][InputT] and return a
+ * [rendering][RenderingT]. To that end, a workflow may recursively ask other workflows to render
+ * themselves, subscribe to data streams from the outside world, and handle events both from its
+ * [renderings][WorkflowContext.onEvent] and from workflows it's delegated to (its "children"). A
+ * `Workflow` may also emit [output events][OutputT] up to its parent `Workflow`.
+ *
+ * Workflows form a tree, where each workflow can have zero or more child workflows. Child workflows
+ * are started as necessary whenever another workflow asks for them, and are cleaned up automatically
+ * when they're no longer needed. [Input][InputT] propagates down the tree, [outputs][OutputT] and
+ * [renderings][RenderingT] propagate up the tree.
+ *
+ * @param InputT Typically a data class that is used to pass configuration information or bits of
+ * state that the workflow can always get from its parent and needn't duplicate in its own state.
+ * May be [Unit] if the workflow does not need any input data.
+ *
+ * @param OutputT Typically a sealed class that represents "events" that this workflow can send
+ * to its parent.
+ * May be [Nothing] if the workflow doesn't need to emit anything.
+ *
+ * @param RenderingT The value returned to this workflow's parent during [composition][compose].
+ * Typically represents a "view" of this workflow's input, current state, and children's renderings.
+ * A workflow that represents a UI component may use a view model as its rendering type.
+ *
+ * @see StatefulWorkflow
  */
-typealias StatelessWorkflow<InputT, OutputT, RenderingT> =
-    Workflow<InputT, Unit, OutputT, RenderingT>
+abstract class StatelessWorkflow<InputT : Any, OutputT : Any, RenderingT : Any> :
+    Workflow<InputT, OutputT, RenderingT> {
+
+  /**
+   * Called at least once any time one of the following things happens:
+   *  - This workflow's [input] changes (via the parent passing a different one in).
+   *  - A child workflow updates in some way (e.g. a stateful child's state changes).
```

Comment:
s/_(e.g.…)/(i.e., it emits output or updates its rendering)/

And add a bullet about workers?

Reply:
What would you like to see about workers? That term doesn't exist in any of the v2 code yet. I've updated this comment to match the updated wording for `StatefulWorkflow`.

Reply:
_ That term doesn't exist in any of the v2 code yet_ Excellent.